infobserve/common/queue.py

```diff
@@ -93,11 +93,13 @@ async def wait_all(self):
         has been called.
         """
         if self.type == self.REDIS_QUEUE:
-            pass
+            with await self.__queue.redis as conn:
+                redis = Redis(conn)
+                return await redis.llen(self.name)
         else:
             await self.__queue.join()
 
-    def events_left(self):
+    async def events_left(self):
```

Comment:
Why does this need to be async?

Reply:
because i am blind i added the logic in `wait_all` method :face_with_head_bandage: